prow/config/agent.go

```diff
@@ -40,10 +41,30 @@ func (ca *Agent) Start(path string) error {
 	}
 	ca.c = c
 	go func() {
-		for range time.Tick(1 * time.Minute) {
+		var lastModTime time.Time
+		// Rarely, if two changes happen in the same second, mtime will
+		// be the same for the second change, and an mtime-based check would
+		// fail. Reload periodically just in case.
+		skips := 0
+		for range time.Tick(1 * time.Second) {
+			if skips < 600 {
+				// Check if the file changed to see if it needs to be re-read.
+				// os.Stat follows symbolic links, which is how ConfigMaps work.
+				stat, err := os.Stat(path)
+				if err != nil {
+					logrus.WithField("path", path).WithError(err).Error("Error loading config.")
+					continue
+				}
+				if stat.ModTime() == lastModTime {
```

Comment:
https://golang.org/pkg/time/#Time.Equal

Reply:
we should probably make this change..

Reply:
It's irrelevant for this case.

Reply:
per the docs it won't matter in this case, but it still smells a little. time.Time has unexported fields so we should probably use `Equal`, I don't think there's any guarantee preventing a change to the internals that break this other than it would probably break a lot of code :-)